### ISSUE TITLE
Conflict with Django report builder

### ISSUE THREAD

**Issue by kunalgrover05** [noripyt/django-cachalot#110]
Cachalot seems to break [Django-report-builder's](https://github.com/burke-software/django-report-builder) functionality.
Adding stack trace here:

```
Internal Server Error: /report_builder/api/report/2/generate/
Traceback (most recent call last):
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/rest_framework/views.py", line 494, in dispatch
    response = self.handle_exception(exc)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/rest_framework/views.py", line 454, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/rest_framework/views.py", line 491, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/report_builder/api/views.py", line 283, in get
    return self.post(request, report_id=report_id)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/report_builder/api/views.py", line 290, in post
    preview=True,)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/report_builder/models.py", line 244, in report_to_list
    for obj in queryset:
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/db/models/query.py", line 250, in __iter__
    self._fetch_all()
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/db/models/query.py", line 1118, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/cachalot/monkey_patch.py", line 32, in inner
    return original(compiler, *args, **kwargs)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/cachalot/monkey_patch.py", line 81, in inner
    cache_key, table_cache_keys)
  File "/home/kunal/leanagri/env3/lib/python3.6/site-packages/cachalot/monkey_patch.py", line 46, in _get_result_or_execute_query
    timestamp, result = data.pop(cache_key)
TypeError: 'NoneType' object is not iterable
```

Any suggestions on what might be the possible conflict to look into?

**Comment by kunalgrover05** [comment]
Value in the dictionary is None.

https://github.com/noripyt/django-cachalot/blob/74e000b8e2b7f4cf9f31319a83d21620ea7b747d/cachalot/monkey_patch.py#L45-L48

**Comment by BertrandBordage** [comment]
Which cache backend are you using?

**Comment by kunalgrover05** [comment]
I am using Memcache backend, but I don't have any Cache setup currently in my Development environment.

**Comment by kunalgrover05** [comment]
I ran memcached on my development environment and that fixed it.

I have never encountered this issue otherwise. Any ideas on how to fix it?
I believe it is possibly dangerous since Memcache server being unavailable would cause 500s on my application.

**Comment by BertrandBordage** [comment]
Could you share your exact `CACHES` configuration please?

**Comment by kunalgrover05** [comment]
Here it is

```python
    CACHES = {
        'default': {
            'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
            'LOCATION': '127.0.0.1:11211',
            'KEY_PREFIX': os.environ['VERSION']
        },
        'debug-panel': {
            'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
            'LOCATION': '/var/tmp/debug-panel-cache',
            'OPTIONS': {
                'MAX_ENTRIES': 200
            }
        }
    }
```

Again note that cachalot works flawlessly unless I click on the Generate report of Django report builder.

**Comment by kunalgrover05** [comment]
UPDATE: <s>Again note that cachalot works flawlessly unless I click on the Generate report of Django report builder.</s>

Fails at unrelated pieces too. Any ideas on how to further debug it?

**Comment by kunalgrover05** [comment]
Any ideas? It will be really unfortunate to stop using cachalot :(

**Comment by kunalgrover05** [comment]
Simply trying to circumvent the problem: https://github.com/noripyt/django-cachalot/compare/master...kunalgrover05:master

The above works, I will be able to report based on the logs I receive from my application and if it is something we can find and fix.

**Comment by BertrandBordage** [comment]
Fixed by 2.0.2.
